packages/components/src/internal/components/administration/GroupManagement.spec.tsx

```diff
@@ -0,0 +1,71 @@
+import React from 'react';
+
+import { List, Map } from 'immutable';
+
+import { TEST_PROJECT, TEST_PROJECT_CONTAINER } from '../../../test/data/constants';
+import { mountWithAppServerContext, waitForLifecycle } from '../../testHelpers';
+import { createMockWithRouteLeave } from '../../mockUtils';
+import { getSecurityTestAPIWrapper, SecurityAPIWrapper } from '../security/APIWrapper';
+import { AppContext } from '../../AppContext';
+import { getTestAPIWrapper } from '../../APIWrapper';
+import { SecurityPolicy } from '../permissions/models';
+import policyJSON from '../../../test/data/security-getPolicy.json';
+import { ServerContext } from '../base/ServerContext';
+import { TEST_USER_FOLDER_ADMIN } from '../../userFixtures';
+
+import { GroupManagement, GroupManagementImpl, GroupPermissionsProps } from './GroupManagement';
+
+describe('BasePermissions', () => {
+    function getDefaultProps(): GroupPermissionsProps {
+        return {
+            error: undefined,
+            inactiveUsersById: Map(),
+            principals: List(),
+            principalsById: Map(),
+            roles: List(),
+            rolesByUniqueName: Map(),
+            ...createMockWithRouteLeave(jest.fn),
+        };
+    }
+
+    const TEST_POLICY = SecurityPolicy.create(policyJSON);
+
+    function getDefaultAppContext(overrides?: Partial<SecurityAPIWrapper>): Partial<AppContext> {
+        return {
+            api: getTestAPIWrapper(jest.fn, {
+                security: getSecurityTestAPIWrapper(jest.fn, {
+                    fetchPolicy: jest.fn().mockResolvedValue(TEST_POLICY),
+                    fetchGroups: jest.fn().mockResolvedValue([]),
+                    getGroupMemberships: jest.fn().mockResolvedValue([]),
+                    ...overrides,
+                }),
+            }),
+        };
+    }
+
+    function getDefaultServerContext(): Partial<ServerContext> {
+        return {
+            container: TEST_PROJECT_CONTAINER,
+            project: TEST_PROJECT,
+            user: TEST_USER_FOLDER_ADMIN,
+        };
+    }
+
+    test('loads', async () => {
+        const wrapper = mountWithAppServerContext(
+            <GroupManagementImpl {...getDefaultProps()} />,
+            getDefaultAppContext(),
+            getDefaultServerContext()
+        );
+
+        expect(wrapper.find(GroupManagement).exists()).toBe(false);
+        await waitForLifecycle(wrapper);
+        // expect(wrapper.find(GroupManagement).exists()).toBe(true);
```

Comment:
Because there is a call to a selectRows() in the useEffect() of GroupManagement.tsx (see getAuditLogData()) the page stays in a loading state. I can see endpoint payload mocks, but is there a way to similarly mock a selectRows() call?

Reply:
The way to handle this would be to add `getAuditLogData` to one of the APIWrappers so that it could be mocked in the `getTestAPIWrapper` like the `fetchPolicy` and `getGroupMemberships` above.  Then in the `GroupManagement.tsx` component at line 64, the call would be changed to `api.security.getAuditLogData`